src/Airship/Internal/Decision.hs

```diff
@@ -559,12 +559,17 @@ l13 r@Resource{..} = do
         Nothing ->
             m16 r
 
-l07 r = do
+l07 r@Resource{..} = do
     trace "l07"
     req <- lift request
-    if requestMethod req == HTTP.methodPost
-        then m07 r
-        else lift $ halt HTTP.status404
+    if requestMethod req == HTTP.methodPost then m07 r
+    else do
+        m <- _contentType <$> get
+        when (isJust m) $ do
+            let (Just (ct, body)) = m
+            lift body >>= lift . putResponseBody
+            lift $ addResponseHeader ("Content-Type", renderHeader ct)
```

Comment:
> I wonder if rather than a Resource that is only intended for 404 responses if it might be better to accept some mapping of status codes to responses

FWIW I think that something that solves this is more useful generally. If you have a route that matches a path, but isn't in the database (for example) you want the same behaviour. Or not having permissions (403). etc. For now all you can do is manually put the responseBody before returning a `Bool`, but I feel like the types could be strong to enforce/allow for `ResponseBody` (based on the `Accept` even) to be returned if/when you're halting.